### ISSUE TITLE
test/configure.ac broken on macOS

### ISSUE THREAD

**Issue by icculus** [libsdl-org/SDL#6494]
Trying to configure the tests on macOS fails now, because there's no pkg-config, I assume.

This broke in 64c93f7804196e74e365570a01f25bfd2c0199b4; reverting it fixes the issue, but that's probably not the solution we're looking for.

Relevant console output:

```
checking for pkg-config... no
checking for sdl2 >= 2.0.18... no
configure: error: *** SDL version 2.0.18 not found!
```

(CC @sezero)

**Comment by icculus** [comment]
(latest from SDL's git is installed on this system, build with the autoconf script and "sudo make install" to /usr/local, fwiw.)

**Comment by slouken** [comment]
Yes. I'm also wondering if we should back out @smcv's commit as well, since there are environments that don't provide pkg-config by default.

This will be worth revisiting in SDL3, if we drop autoconf support entirely.

**Comment by sezero** [comment]
> Yes.

I guess that is directed at me - will revert 64c93f7 shortly.  Should I revert the corresponding commits from satellite libs too?

**Comment by sezero** [comment]
I went ahead and reverted the commit in here, and also in all of the satellite libs. Closing.

**Comment by eli-schwartz** [comment]
> Yes. I'm also wondering if we should back out @smcv's commit as well, since there are environments that don't provide pkg-config by default.

It's pretty easy to install, though, on any platform other than "Windows without mingw" (which is unlikely to use autotools) and is at least possible to install even there.

So, for exactly the reason mentioned in #6467, it is probably a good idea to use it as it's much more reliable -- but if you really want to avoid SDL detection failure when pkg-config is missing, then maybe you could at least try PKG_CHECK_MODULES, and instead of AC_MSG_ERROR if it fails, just fall back to AM_PATH_SDL2?

**Comment by slouken** [comment]
> So, for exactly the reason mentioned in #6467, it is probably a good idea to use it as it's much more reliable -- but if you really want to avoid SDL detection failure when pkg-config is missing, then maybe you could at least try PKG_CHECK_MODULES, and instead of AC_MSG_ERROR if it fails, just fall back to AM_PATH_SDL2?

That seems like a good approach, although we're discussing switching to CMake for SDL3, so this might be a moot point soon.

**Comment by sezero** [comment]
> So, for exactly the reason mentioned in #6467, it is probably a good idea to use it as it's much more reliable -- but if you really want to avoid SDL detection failure when pkg-config is missing, then maybe you could at least try PKG_CHECK_MODULES, and instead of AC_MSG_ERROR if it fails, just fall back to AM_PATH_SDL2?

AM_PATH_SDL2 already does that internally, does it not?

**Comment by smcv** [comment]
> AM_PATH_SDL2 already does that internally, does it not?

Sort of, it's a bit more complicated than that (it can use `PKG_CHECK_MODULES` internally or not, depending on options).

Half of the point of #6467 is that we want to encourage third-party users of SDL to stop using `AM_PATH_SDL2`, as an incremental step towards SDL 3 not providing an equivalent at all. That implies encouraging those third-party users to stop relying on *anything* that `AM_PATH_SDL2` does.

Is it a requirement that the `test/` subdirectory continues to be compilable against a system-installed SDL? A more common approach to automated/manual tests in the same source tree as the project under test is to compile them as part of the main project (with a build flag to enable or disable them), more like the way SDL_image works. If SDL's `test/` was more like that, then it wouldn't need to discover SDL at all: it could assume that SDL is the just-compiled copy in the parent directory, and set compile/link flags accordingly.

**Comment by smcv** [comment]
> AM_PATH_SDL2 already does that internally, does it not?

Sort of, it's a bit more complicated than that (it can use `PKG_CHECK_MODULES` internally or not, depending on options).

Half of the point of #6467 is that we want to encourage third-party users of SDL to stop using `AM_PATH_SDL2`, as an incremental step towards SDL 3 not providing an equivalent at all. That implies encouraging those third-party users to stop relying on *anything* that `AM_PATH_SDL2` does.

Is it a requirement that the `test/` subdirectory continues to be compilable against a system-installed SDL? A more common approach to automated/manual tests in the same source tree as the project under test is to compile them as part of the main project (with a build flag to enable or disable them), more like the way SDL_image works. If SDL's `test/` was more like that, then it wouldn't need to discover SDL at all: it could assume that SDL is the just-compiled copy in the parent directory, and set compile/link flags accordingly.

**Comment by slouken** [comment]
No, the test subdirectory doesn't need to build against the system installed SDL, it can build against the in-tree version, it's just not set up to do that right now.

Let's not change that now, but we'll revisit with build tools decisions in 3.0
